### PR TITLE
AddFstEntriesToGameList - prevent loading a directory as a file

### PR DESCRIPTION
**Pull request by LFsWang**
truth table
image

before
old

after
new

### CODE REVIEW

File: src/citra_qt/game_list.cpp [hunk 1]
```diff
@@ -145,6 +143,8 @@ void GameListWorker::AddFstEntriesToGameList(const std::string& dir_path, unsign
                 new GameListItem(QString::fromStdString(Loader::GetFileTypeString(loader->GetFileType()))),
                 new GameListItemSize(FileUtil::GetSize(physical_name)),
             });
+        } else if (recursion > 0) {
```
Comment (LittleWhite-tb) [inline src/citra_qt/game_list.cpp]: I would add a small comment here to specify that physical_name is a directory here. Maybe I am over commenting... so do as you like.

**Comment by LittleWhite-tb** [comment]
Everything fine to me :+1:

File: src/citra_qt/game_list.cpp (hunk 1 above)
Comment (lioncash) [inline src/citra_qt/game_list.cpp]: The if statement kind of makes it evident that it's a directory at this point.

Comment (LittleWhite-tb) [inline src/citra_qt/game_list.cpp]: Sure, maybe I'm a bit sleepy :)

**Comment by bunnei** [comment]
Lgtm

**Pull request merged**